backend/dataall/modules/shares_base/services/sharing_service.py

```diff
@@ -356,6 +362,48 @@ def reapply_share(cls, engine: Engine, share_uri: str) -> bool:
                     if not lock_released:
                         log.error(f'Failed to release lock for dataset {share_data.dataset.datasetUri}.')
 
+    @staticmethod
+    def fetch_pending_shares(engine):
+        """
+        A method used by the scheduled ECS Task to run fetch_pending_shares() process against ALL shared objects in ALL
+        active share objects within dataall
+        """
+        with engine.scoped_session() as session:
+            pending_shares = (
+                session.query(ShareObject)
+                .join(
+                    Notification,
+                    and_(
+                        ShareObject.shareUri == func.split_part(Notification.target_uri, '|', 1),
+                        ShareObject.datasetUri == func.split_part(Notification.target_uri, '|', 2),
+                    ),
+                )
+                .filter(and_(Notification.type == 'SHARE_OBJECT_SUBMITTED', ShareObject.status == 'Submitted'))
+                .all()
+            )
+            return pending_shares
+
```

Comment:
The SharingService was conceived to manage actual AWS calls to shares (verify, approve...) share_notifications_emails do not really fit into this category. I think we can simplify the implementation by moving the logic of `_get_share_data` directly to the task and call the ShareNotificationService directly.  We would remove all the code in SharingService and have a task looking something like this:

```
def persistent_email_reminders(engine, envname):
    """
    A method used by the scheduled ECS Task to run persistent_email_reminder() process against ALL
    active share objects within data.all and send emails to all pending shares.
    """
    with engine.scoped_session() as session:
        log.info('Running Persistent Email Reminders Task')
        pending_shares = SharingService.fetch_pending_shares(engine=engine)
        log.info(f'Found {len(pending_shares)} pending shares')
        pending_share: ShareObject
        for pending_share in pending_shares:
            log.info(f'Sending Email Reminder for Share: {pending_share.shareUri}')
            SharingService.persistent_email_reminder(uri=pending_share.shareUri, envname=envname)
        log.info('Completed Persistent Email Reminders Task')


```

Reply:
```
    @staticmethod
    @Worker.handler(path='ecs.share.persistent_email_reminder')
    def persistent_email_reminder(engine, task: Task):
        return EcsShareHandler._manage_share(
            engine, task, SharingService.persistent_email_reminder, 'persistent_email_reminder'
        )
```
`SharingService.persistent_email_reminder` is used in ecs_share_handler.py like this. I like your suggestion to move it out of sharing service, but how would we define the ecs task in the ecs_share_handler.py then?